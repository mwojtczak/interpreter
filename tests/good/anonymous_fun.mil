function test (var a, func) {
	print (func(a, 41));
	return func(a, 42);
}

function main () {
	a = 4;
	i = 20;
	j = 10;
	print(test (a, function (var i, val j) {return(i + j);}));

	apply = function (f, var x, val y){ 
			return f(x, y);
		};
	exec_and_apply = function (p, f, var x, var y) {
		p(x,y);
		return(apply(f, x, y));
	};
	print(apply(function (val x, val y){return(x + y);}, i, j));
	print(apply(function (val x, val y){return(x - y);}, i, j));
	print(apply(function (val x, val y){return(x * y);}, i, j));
	print(apply(function (val x, val y){return(x / y);}, i, j));
	print(apply(function (val x, val y){return(x < y);}, i, j));
	print(apply(function (val x, val y){return(x > y);}, i, j));
	
	print(apply(function (var x, var y) {
			print(i);
			i += 20;
			print(i);
			print(x);
			return(x + y);
		}, i, j))

	print(exec_and_apply(function (var x, var y){
			print(x);
			print(y);
			print(i);
			i += 10;
		}, function (var x, val y){return(x + y);}, i, j));
	print(i); 
}
