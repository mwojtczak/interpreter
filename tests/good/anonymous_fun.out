45
46
30
10
200
2
false
true
20
40
40
50
40
10
40
60
50
